module/zcommon/zfs_prop.c

```diff
@@ -609,6 +625,24 @@ zfs_prop_init(void)
 	    ZFS_TYPE_DATASET | ZFS_TYPE_BOOKMARK, "<snapshot>[,...]",
 	    "RSNAPS", sfeatures);
 
+	/* __APPLE__ */
+	zprop_register_index(ZFS_PROP_BROWSE, "com.apple.browse", 1,
```

Comment:
We haven't got any properties prefixed like this, even the platform-specific ones like `jailed` and `zoned`. I wonder if these should just lose the prefix? And, maybe, have an `#ifdef`?

If they must keep the prefix, I'd suggest it be something that we control, maybe `org.openzfs:macos_browse`. But I don't really see why.

Last guess: if they are named this because they're plumbed through from some proper macOS feature (do they have a notion of filesystem properties?) then I'd at least add a comment explain that, though more likely I'd do it with a mapping table closer to the edge.

Also, the casing and short names are inconsistent through these properties.

Reply:
I haven't removed the prefix
I've added an ifdef
I've cleaned up the case
I haven't cleaned up the shortname inconstancy

Reply:
@lundman do you plumb them through anywhere, or should I just remove the prefix?

Reply:
I talked to @lundman and he said that "com.apple.browse", "com.apple.ignoreowner", "COM.APPLE.LASTUNMOUNT", "com.apple.mimic", "com.apple.devdisk" are needed for backwards compatability with current zfs on mac pools, since the string is stored on disk.

Reply:
Fair enough! In that case I'd probably just put a comment in to say why they look weird.

(I mean, we could do a whole thing to recognise the old ones and upgrade them to "preferred" ones, but its probably not worth the maintenance overhead).

Reply:
Since we were the first, at the time anyway, of adding platform specific properties, we took the approach suggested by the ZFS manual pages for user-defined properties, like `zfs set com.delphix:backup=true tank`. But then it appears that FreeBSD just go with the property name, so I suppose that was a waste.
I don't mind what they are called, but if they are changed, older macOS pools will not understand the setting, and users can set it again. That is fine too.

BROWSE and IGNOREOWNER matches macOS mount specific options. 
LASTMOUNT is that Spotlight needs to know when it was last mounted.
MIMIC is to pretend to be "hfs" instead of "zfs", since far too many apps just go `ismount(fs=="hfs")` instead of using the exceptionally detailed capabilities API. Adobe, and Apple included.
DEVDISK is about if we wrap each mount in a pseudo /dev/disk, as required by the UI, but gets cluttered for all internal datasets. Similar to that freebsd zvol one for geom right?

Reply:
If you do add #ifdef __APPLE__ - then you will not be able to change the setting on a non-macOS platform, say under recovery situations. But perhaps that is not needed. If other platforms have #ifdef around their specific features, then lets do the same.

Reply:
I'm not strongly opinionated either way, but if its no problem to just use no prefix, then I'd do that. But I think keeping them for backcompat is ok too.

(If I wanted a prefix, it wouldn't be `com.apple` because we're not Apple (I don't think so!), but it gets in the way of user-defined properties anyway).

Thanks for explaining what they're for; I am not very macOS-aware these days!

Reply:
Yeah, it's awkward to use com.apple., we had long discussions about it at the the time (2013?), good thing we didnt go with OsX, since that is no longer around. There are xattrs that are named starting with `com.apple.` that we have to use to be compatible with the system, so we took it as the "OS made by apple". But dropping the prefix would probably work, and put a note in the upgrade documentation.

Reply:
@lundman are you able to do that?

Reply:
Just make
```
zprop_register_index(ZFS_PROP_BROWSE, "com.apple.browse", 1,
```
say
```
zprop_register_index(ZFS_PROP_BROWSE, "browse", 1,
```

and we'll have to make sure the documentation is up to date, since everyone will wonder what
"browse" is in the dataset properties, when before they could ignore it when they see the "apple" part :)

Reply:
How about "apple_force_mntopt_browse"

Reply:
Done